authentication/http.go

```diff
@@ -27,35 +25,25 @@ const (
 	tenantIDKey contextKey = "tenantID"
 )
 
-// WithTenant finds the tenant from the URL parameters and adds it to the request context.
-func WithTenant(next http.Handler) http.Handler {
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		tenant := chi.URLParam(r, "tenant")
-		next.ServeHTTP(w, r.WithContext(
-			context.WithValue(r.Context(), tenantKey, tenant),
-		))
-	})
-}
-
-// WithTenantID returns a middleware that finds the tenantID using the tenant
-// from the URL parameters and adds it to the request context.
-func WithTenantID(tenantIDs map[string]string) Middleware {
+// WithTenant returns a middleware with request context
+// containing both ID and name of a tenant.
+func WithTenant(tenantName, tenantID string) Middleware {
```

Comment:
I'm just thinking why we need `tenantName`, shouldn't it be available from route context via `chi.URLParam` as it was before?

Reply:
ahh! two reasons:

1. there was quite a bit confusion / concerns on URLParam can be manipulated vs. URLParam is implicitly gets added by 'Chi'
2. the Chi's curly notation ( that Chi uses to populate chi.URLParam automatically from the incoming request) was not working with multiple Chi mounts.